homeassistant/components/esphome/__init__.py

```diff
@@ -114,8 +114,17 @@ def async_on_service_call(service: ServiceCall) -> None:
                 _LOGGER.error('Error rendering data template: %s', ex)
                 return
 
-        hass.async_create_task(hass.services.async_call(
-            domain, service_name, service_data, blocking=True))
+        if service.is_event:
+            # ESPHome uses servicecall packet for both events and service calls
+            # Ensure the user can only send events of form 'esphome.xyz'
+            if domain != 'esphome':
+                raise ValueError("Can only generate events under esphome "
+                                 "domain!")
+            hass.bus.async_fire(service.service, service_data,
+                                EventOrigin.remote)
```

Comment:
The event origin is only used if another home assistant instance is the origin of the event (deprecated). So now we always have the origin be local, which is the default.

Reply:
Ok, good to know. Looks like deconz and mobile_app need to be updated in that case as well.